src/legacy/core_plugins/kibana/public/dashboard/dashboard_app_controller.tsx

```diff
@@ -234,6 +257,15 @@ export class DashboardAppController {
         if (!isErrorEmbeddable(container)) {
           dashboardContainer = container;
 
+          dashboardContainer.renderEmpty = () => {
+            const shouldShowEditHelp = $scope.getShouldShowEditHelp();
+            const shouldShowViewHelp = $scope.getShouldShowViewHelp();
+            const isEmptyState = shouldShowEditHelp || shouldShowViewHelp;
```

Comment:
It seems like those two functions are always used like this. Would it make sense to put them into a single function and just call it directly here?

Reply:
No, there is distinction between `edit` and `view` screen, it's used in the `getEmptyScreenProps` to decide which screen should be rendered.